packages/api/amplify_api_dart/lib/src/graphql/factories/graphql_request_factory.dart

```diff
@@ -70,6 +70,19 @@ class GraphQLRequestFactory {
           ignoreParents: true,
         ); // always format like a get, stop traversing parents
         fields.add('${belongsTo.name} { $parentSelectionSet }');
+
+        // Include the ID key(s) itself in selection set. This is ignored while
+        // deserializing response because model will use ID nested in `parentSelectionSet`.
+        // However, including the ID in selection set allows subscriptions that
+        // filter by these ID to be triggered by these mutations.
+        var belongsToKeys = belongsTo.association?.targetNames;
+        if (belongsToKeys == null &&
+            belongsTo.association?.targetName != null) {
+          belongsToKeys = [belongsTo.association?.targetName as String];
+        }
+        if (belongsToKeys != null) {
+          fields.addAll(belongsToKeys);
```

Comment:
Question: Is this for when a Model has multiple belongsTo relationships?

Reply:
This is actually for if there is a custom primary key w complex identifier (multiple keys) on a single relationship. Even though the CPK part not merged to target branch yet, just wrote this in a way will work with CPK, or simpler, older codegens.

Reply:
Q: does this change interfear with `buildInputVariableForMutations`?

Reply:
No, will not interfere with that. We are already including these IDs in the input object built by `buildInputVariableForMutations` as the input of the mutation, this PR just a small change to requested response from AppSync.